### PR TITLE
Feature: incremental order-book

### PR DESCRIPTION
**Pull request by calj**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo_slug| "https://github.com/#{repo_slug}" }
 
-ruby '2.6.5'
+ruby '~> 2.6'
```
Comment (ysv) [inline Gemfile:7]: do we need this change ?

File: app/trading/matching/price_level.rb [hunk 2]
```diff
@@ -25,6 +25,10 @@ def add(order)
       end
     end
 
+    def total
+      @orders.map(&:volume).reduce(ZERO, :+)
```
Comment (ysv) [inline app/trading/matching/price_level.rb]: I would prefer using `#sum`

**Review by ysv — COMMENTED** [review]
The only design inaccuracy I see is:
Is it ok that matching will be broken in case `Peatio::Ranger::Events.publish` errors ?

File: Gemfile (hunk 1 above)
Comment (calj) [reply Gemfile:7]: I have 2.6.3 locally, do we need to force developers to upgrade their ruby?
For me the important version is the one in Dockerfile.

**Comment by calj** [comment]
Exceptions in ranger publish are rescued now

**Comment by calj** [comment]
@mnaichuk I think this should be tagged 2.4

**Pull request merged**